src/charm.py

```diff
@@ -192,10 +198,11 @@ def __init__(self, *args):
     @property
     def self_scraping_job(self):
         """The self-monitoring scrape job."""
-        port = urlparse(self._external_url).port or 80
+        external_url = urlparse(self._external_url)
         return [
             {
-                "static_configs": [{"targets": [f"*:{port}"]}],
+                "metrics_path": f"{external_url.path}/metrics",
+                "static_configs": [{"targets": [f"{external_url.hostname}:{external_url.port}"]}],
```

Comment:
Iiuc, we don't need IPU to scrape all units, because peer addresses are [already in relation data](https://github.com/canonical/alertmanager-k8s-operator/blob/bbb5f539795c481aa47ff8ac97053a62173b846a/src/charm.py#L575).

We probably could construct the `targets` list to include all known peers?

Reply:
Let's follow this in: https://github.com/canonical/alertmanager-k8s-operator/issues/151